RubberduckBaseProject.csproj

```diff
@@ -72,6 +72,60 @@
   </Target>
+
+  <ItemGroup>
+    <CodeAnalyzersToBuild Include="$(SolutionDir)\RubberduckCodeAnalysis\RubberduckCodeAnalysis.csproj">
+      <Properties>Configuration=Release</Properties>
+    </CodeAnalyzersToBuild>
+    <MSBuildProjectsToBuild Include="$(SolutionDir)\Rubberduck.Deployment.Build\Rubberduck.Deployment.Build.csproj">
+      <Properties>Configuration=Release</Properties>
+    </MSBuildProjectsToBuild>
+    <MetaTestsToBuild Include="$(SolutionDir)\RubberduckTestsCodeAnalysis\RubberduckTestsCodeAnalysis.csproj">
+      <Properties>Configuration=Release</Properties>
+    </MetaTestsToBuild>
+  </ItemGroup>
+  <Target Name="EnsureCodeAnalyzersBuild" BeforeTargets="PrepareForBuild"
+          Condition="!Exists('$(SolutionDir)RubberduckCodeAnalysis\bin\RubberduckCodeAnalysis.dll')">
+    <Message Text="Missing the required code analyzers. Building the code analzyers..." Importance="high" />
+    <MSBuild Projects="@(CodeAnalyzersToBuild)"
+             BuildInParallel="false"
+             ContinueOnError="false"
+             SkipNonexistentProjects="false"
+             StopOnFirstFailure="true"
+             UnloadProjectsOnCompletion="true"
+             UseResultsCache="false" 
+             RunEachTargetSeparately=""
+             RebaseOutputs="true"/>
+    <Message Text="Built the code analyzers" Importance="high" />
+  </Target>
+  <Target Name="EnsureMSBuildTargetBuild" BeforeTargets="PrepareForBuild"
+          Condition="!Exists('$(SolutionDir)Rubberduck.Deployment.Build\bin\Rubberduck.Deployment.Build.dll')">
+    <Message Text="Missing the required build tasks. Building the build tasks..." Importance="high" />
+    <MSBuild Projects="@(MSBuildProjectsToBuild)"
+             BuildInParallel="false"
+             ContinueOnError="false"
+             SkipNonexistentProjects="false"
+             StopOnFirstFailure="true"
+             UnloadProjectsOnCompletion="true"
+             RunEachTargetSeparately="true"
+             UseResultsCache="true"
+             RebaseOutputs="true"/>
+    <Message Text="Built the build tasks" Importance="high" />
+  </Target>
+  <Target Name="EnsureMetaTestsBuild" BeforeTargets="PrepareForBuild"
+          Condition="!Exists('$(SolutionDir)RubberduckTestsCodeAnalysis\bin\RubberduckTestsCodeAnalysis.dll')">
+    <Message Text="Missing the required meta tests. Building the meta tests..." Importance="high" />
+    <MSBuild Projects="@(MetaTestsToBuild)"
+             BuildInParallel="false"
+             ContinueOnError="false"
+             SkipNonexistentProjects="false"
+             StopOnFirstFailure="true"
+             UnloadProjectsOnCompletion="true"
+             RunEachTargetSeparately="true"
+             UseResultsCache="true"
+             RebaseOutputs="true"/>
+    <Message Text="Built the build tasks" Importance="high" />
+  </Target>
```

Comment:
This is basically mirroring the structure of the Meta solution. Could we build that meta solution instead of it's composing projects?

Reply:
In fact that's not what I originally wanted, either. I had started out with a single `MSBuild` task to build the `RubberduckMeta.sln`. 

In spite of [the documentation](http://msdn.microsoft.com/en-us/library/z7f65y0d.aspx) and [other people reporting success](https://stackoverflow.com/a/7096698/643342), using this instead:

```
  <ItemGroup>
    <MetaProjectsToBuild Include="$(SolutionDir)\RubberduckMeta.sln">
      <Properties>Configuration=Release</Properties>
    </MetaProjectsToBuild>
  </ItemGroup>
  <Target Name="EnsureMetaProjectsBuild" BeforeTargets="PrepareForBuild"
          Condition="!Exists('$(SolutionDir)RubberduckCodeAnalysis\bin\RubberduckCodeAnalysis.dll')">
    <Message Text="Missing the required meta projects. Building the meta projects..." Importance="high" />
    <MSBuild Projects="@(MetaProjectsToBuild)"
             BuildInParallel="false"
             ContinueOnError="false"
             SkipNonexistentProjects="false"
             StopOnFirstFailure="true"
             UnloadProjectsOnCompletion="true"
             UseResultsCache="false"
             RunEachTargetSeparately="true"
             RebaseOutputs="true"/>
    <Message Text="Built the meta projects" Importance="high" />
  </Target>
```

The messages about missing metaprojects and have built them do come up... but the projects simply don't get build. WTF? 

Only by specifying them on a per-project, could I get them built. Maybe I'm doing something wrong...

Reply:
Had few attempts but I could not get it to actually build using a solution reference. For now, I commented out my last attempt which seemed closest (but still doesn't build) and enumerate the projects. At least we only define the `MSBuild` task once now that we are passing in an array of projects.